### ISSUE TITLE
[bug?] Extreme color variations in txt2img

### ISSUE THREAD

**Issue by Kanareika** [Mikubill/sd-webui-controlnet#373]
sd-webui-controlnet: latest commit at the time of writing
I encountered a strange problem today. When using the hed, canny, or scribble, the resulting image will have too saturated colors. It depends on "weight" slider, i.e. the smaller the value, the colors more correct. No difference between control_sd15_ and difference_controlsd15 models. Effect is extremely noticeable on realistic models, less - on anime models. Example:
Original image (in ControlNet):
![08514-1328727690-1girl, solo, a photo of (beautiful girl standing in the ruins), (wearing a rusty used power armor_1 2), weary sci-fi powersuit,](https://user-images.githubusercontent.com/10144329/221362563-cb3f45a5-a58f-4ca2-a4b6-0b8ed6e3df95.png)
heg, weight 1:
![08548-3421591712-1girl, solo, a photo of (beautiful girl standing in the ruins), (wearing a rusty used power armor_1 2), weary sci-fi powersuit,](https://user-images.githubusercontent.com/10144329/221362592-7df93eef-38cd-4cea-9da1-d6bd097228fd.png)
heg, weight 0.6:
![08549-3421591712-1girl, solo, a photo of (beautiful girl standing in the ruins), (wearing a rusty used power armor_1 2), weary sci-fi powersuit,](https://user-images.githubusercontent.com/10144329/221362636-612a6082-c063-4231-aca0-b69fe705f862.png)
heg, weight 0.3:
![08550-3421591712-1girl, solo, a photo of (beautiful girl standing in the ruins), (wearing a rusty used power armor_1 2), weary sci-fi powersuit,](https://user-images.githubusercontent.com/10144329/221362647-deb2a69e-2d35-44b2-b36f-17c36a1d3d22.png)
heg, weight 0.1
![08551-3421591712-1girl, solo, a photo of (beautiful girl standing in the ruins), (wearing a rusty used power armor_1 2), weary sci-fi powersuit,](https://user-images.githubusercontent.com/10144329/221362666-0f2e1a53-165b-405f-b4b2-c6cdcb4b7fb5.png)

Any ideas, how to fix it?

**Comment by Kanareika** [comment]
Fixed by enabling **"Only use mid-control when inference"** option. I honestly don't know, what it does, but it's work.

**Comment by gitvirtualmente** [comment]
I tried whit this option, but now it is absolutely different to the photo. I have the same problem.

> Fixed by enabling **"Only use mid-control when inference"** option. I honestly don't know, what it does, but it's work.

**Comment by gitvirtualmente** [comment]
> sd-webui-controlnet: latest commit at the time of writing I encountered a strange problem today. When using the hed, canny, or scribble, the resulting image will have too saturated colors. It depends on "weight" slider, i.e. the smaller the value, the colors more correct. No difference between control_sd15_ and difference_controlsd15 models. Effect is extremely noticeable on realistic models, less - on anime models. Example: Original image (in ControlNet): ![08514-1328727690-1girl, solo, a photo of (beautiful girl standing in the ruins), (wearing a rusty used power armor_1 2), weary sci-fi powersuit,](https://user-images.githubusercontent.com/10144329/221362563-cb3f45a5-a58f-4ca2-a4b6-0b8ed6e3df95.png) heg, weight 1: ![08548-3421591712-1girl, solo, a photo of (beautiful girl standing in the ruins), (wearing a rusty used power armor_1 2), weary sci-fi powersuit,](https://user-images.githubusercontent.com/10144329/221362592-7df93eef-38cd-4cea-9da1-d6bd097228fd.png) heg, weight 0.6: ![08549-3421591712-1girl, solo, a photo of (beautiful girl standing in the ruins), (wearing a rusty used power armor_1 2), weary sci-fi powersuit,](https://user-images.githubusercontent.com/10144329/221362636-612a6082-c063-4231-aca0-b69fe705f862.png) heg, weight 0.3: ![08550-3421591712-1girl, solo, a photo of (beautiful girl standing in the ruins), (wearing a rusty used power armor_1 2), weary sci-fi powersuit,](https://user-images.githubusercontent.com/10144329/221362647-deb2a69e-2d35-44b2-b36f-17c36a1d3d22.png) heg, weight 0.1 ![08551-3421591712-1girl, solo, a photo of (beautiful girl standing in the ruins), (wearing a rusty used power armor_1 2), weary sci-fi powersuit,](https://user-images.githubusercontent.com/10144329/221362666-0f2e1a53-165b-405f-b4b2-c6cdcb4b7fb5.png)
> 
> Any ideas, how to fix it?

I have exactly the same problem. “Only use mid-control when inference” deactivate ControlNet or at least Multi ControlNet. The image is absolutely different.

All my Controlnet images are full saturated and yellow using realistic models. Why? Somebody fixed it?

**Comment by gitvirtualmente** [comment]
After a lot of work I saw that the problem was HED. I don't know why it takes the color of the input image and force to the final output.

**Comment by gitvirtualmente** [comment]
As I see is a BUG

People in general has this problem as we can see in this article:
https://stable-diffusion-art.com/controlnet/#HED

Hope they can fix it.

**Comment by Ericxgao** [comment]
Any updates on this? Still having the same issues here.

**Comment by Mikubill** [comment]
Tested with hed: all results are reproducible in the original controlnet repo and this extension. That means if there is a color shift in the result, it is more likely due to some bias in the original model.

| Base | Extension |
|:-------------------------:|:-------------------------:|
| ![tmpvdvreb5t (1)](https://user-images.githubusercontent.com/31246794/224222634-91747d64-39e2-4bee-b688-2068d1b0c53b.png) | ![00063-81318291 (1)](https://user-images.githubusercontent.com/31246794/224222629-de9ebbb4-e1e1-4b3c-b500-e12a05d325b6.png) |

**Comment by CCpt5** [comment]
Found this thread after posting in a different one here: https://github.com/Mikubill/sd-webui-controlnet/issues/547#issuecomment-1477448158     
  
I get this problem when running CUDA12 and Torch2. Hope the cause is determined.

![00686-3723292620-Morgan_Freeman_giving_a_peace_sign](https://user-images.githubusercontent.com/37972078/226557122-fb4d70c1-8935-47f5-8fee-421d7ff172ab.png)
![5646545646](https://user-images.githubusercontent.com/37972078/226557161-1ffd9aec-adcb-4861-ba3d-5031c9914bdf.jpg)

**Comment by materosx** [comment]
Any update on this? Still have this problem

**Comment by CCpt5** [comment]
> Any update on this? Still have this prob

Do you have Microsoft Visual Studio installed on your system? I did a whole bunch of uninstalling/resinstalling/etc for basically 10hrs a few days ago and I'm pretty in my case the issue was related to Visual Studio. If you did install that previously, try uninstalling it completely and reinstalling it. Then if you have CUDA 11.8 or 12 installed try also uninstalling those and reinstalling them (11.8 is the supported version at the moment).  
  
It's also possible another extension broke something so I removed everything, re-cloned A1111 and started over. I'd tried that first but it didn't fix the problem - so I not 100% sure. If it helps these are the extensions I tended to install even if I used them rarely - I cut it back to just 3 or 4 essentials for me.

ControlNet
Dynamic Prompts
Auto-complete
3-D openpose
Open pose
Remove Background
Ultimate Upscle
Dynamic Thresholding
System Information (this one was the newest I'd installed)

I'd tried other extensions also but generally deleted them fairly quick. Didn't expect problems (and maybe it wasn't them - dunno). There was also a few times w/ A1111 where I checkedout previous versions (commits) and then later returned to the master branch. Reinstalling completely as I did early on I'd expect would have resolved that if it was the fundamental cause, but maybe not....

Good luck!

**Comment by materosx** [comment]
> Good luck!

I dont think it's due to visual studio since I run stable difussion through google colab, but I'll try disabling extensions, other than controlnet I think I only have ultimate sd upscaler and some aspect ratio helper, I'll delete them and see if it helps, thank you

**Comment by AndreyRGW** [comment]
So, still no fix?

**Comment by AndreyRGW** [comment]
> So, still no fix?

I think it might be a problem with the controlnet code or the hed annotator code, 'cause I checked the Feb 26 hed and the Feb 17 hed, they both give the same burned result.

**Comment by vractal** [comment]
this is not only on heed, also get same issue with canny

**Comment by vractal** [comment]
so, are some people not getting this bug? this makes it pretty unusable, so it is probally not everyone or this issue would be filled with people reporting this.. so maybe its a enviroment thing?

**Comment by vractal** [comment]
Interesting, I only get in img2img, and its not always (but almost always)

**Comment by lllyasviel** [comment]
For reference, this problem is solved in CN 1.1. The reason is that the training data of 1.0 has several problems and this image trigged the problem.

Below is results from CN 1.0 (i can reproduce the artifacts)

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/2421fe4d-c7da-4b6b-b021-fc914e35a2fc)

Below is results from CN 1.1 (problem is solved)

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/c114d5f1-ec91-49e7-9860-c190ac0a610b)

The image use the same meta data input with the original image in this post.

**Comment by DESEOUMAIGA** [comment]
> 

Same. Did you find out any way to fix it?

**Comment by tristanvann** [comment]
Bug still exists for me, on CN 1.1.227

**Comment by Imyamoe** [comment]
I started to have this bug on the XL controlnet. I had something similar on a completely unrelated project, after the update on the Community VS. When some certain MSVC library updated broken in FVectors. Instead of the normalized space, they went exponential. So I guess it is may be probably related to some low level library of some sort here as well. But I'm not even sure how to debug it for the Automatic1111. Is anyone is still having this bug yet?
